doc/markdownlinkconverter/markdownlinkconverter.c

```diff
@@ -0,0 +1,452 @@
+		/* [ */			{1, 1, 1, 3},
+		/* ] */			{1, 1, 1, 3},
+		/* blank */		{2, 1, 2, 3},
+		/* \n */		{0, 0, 1, 0},
+		/* ( */			{1, 1, 1, 3},
+		/* ) */			{1, 1, 1, 3},
+		/* # */			{3, 1, 3, 3},
+		/* other*/		{1, 1, 1, 3},
+        }
+    };
+    return out;
+}
+
+// Maps a given char to a int, used for the transition table
+int resolveChar (char c)
+{
+	if (c != '\n' && isblank (c)) return 3;
+	switch (c)
+	{
+		case '!': return 0;
+		case '[': return 1;
+		case ']': return 2;
+		case '\n': return 4;
+		case '(': return 5;
+		case ')': return 6;
+		case '#': return 7;
+		default: return 8; // all other
+	}
+}
+
+/* This function prints out the input to the output, while printing the title
+ * regex is applied. If a title is found a header ("{ #header }") will
+ * be printed out additionally to the output and at the end true will
+ * be returned. If no title is found false will be returned.
+ */
+bool convertTitle (FILE * input, FILE *  output, char * filenameInElektra)
+{
+	int state = TITLE_START;
+	int newstate;
+	char c;
+	bool titleFound = false;
+	struct transitionTitle transitions = genTitleTransitionTable ();
+	while ((c = fgetc (input)) != EOF)
+	{
+		newstate = transitions.t[resolveChar (c)][state];
+
+		if (!titleFound && TITLE_ISGOAL (state, newstate))
+		{
+			titleFound = true;
+			// print Header
+			fprintf (output, " {#");
+			printConvertedPath (output, filenameInElektra);
+			fprintf (output, "}");
+		}
+
+		fprintf (output, "%c", c);
+
+		state = newstate;
+	}
+	return titleFound;
+}
+
+/* This procedure reads from the input and writes to the output, while
+ * processing the file it converts all not Blacklisted links.
+ */
+void convertLinks (FILE * input, FILE * output, char * filenameInElektra)
+{
+	char c;
+	fpos_t pos;
+	int state = LINK_START;
+	int newstate;
+	int index = 0;
+	unsigned int len = 0;
+	/* index and len help to convert a link if found.
+	 *
+	 *  |--index--||-len-|
+	 * [link name](foo/bar)
+	 */
+	struct transitionLink transitions = genLinkTransitionTable ();
+	while ((c = fgetc (input)) != EOF)
+	{
+		newstate = transitions.t[resolveChar (c)][state];
+		if (LINK_POSSIBLE (state, newstate))
+		{
+			// first [, possible link
+			// position is saved for setting back
+			if (fgetpos (input, &pos))
+				exitError (input, NULL, "fgetpos");
+
+			index = 0;
+			len = 0;
+		}
+		if (LINK_FOUND (state, newstate))
+		{
+			//set back and convert link if not blacklisted
+			if (fsetpos (input, &pos))
+				exitError (input, NULL, "fsetpos");
+
+			fprintf (output, "["); //first char got lost
+			//print link name
+			while (index > 0)
+			{
+				fprintf (output, "%c", fgetc (input));
+				--index;
+			}
+			//extract target
+			char target [len * sizeof (char) + 1];
+			if (fread (&target[0], sizeof (char), len, input) != len)
+				exitError (input, NULL, "fread");
+
+			target[len] = '\0';
+
+			//check target
+			bool targetOK = true;
+			//start
+			for (int i = 0;strcmp (ignoreTargetStart[i], "\0") != 0;++i)
```

Comment:
Isn't this the same: `strcmp (ignoreTargetStart[i], "")`? (strcmp should stop at first null-character)

The function gets really long.